File: webapp/frontend/app/src/app/model/conversation/streaming-response.ts
import { LlmMetric } from "./llm-metric";
import { LlmStatus } from "./llm-status";

export const ChunkTypes = [
	'THINKING',
	'RESPONSE',
	'TOOL',
	'STATUS',
	'INTERNAL'
] as const;

export type ChunkType =
	typeof ChunkTypes[number];


export interface StreamingResponse {
	status: LlmStatus,
	metric: LlmMetric,
	sources: string[],
	type: ChunkType,
	content: string
}

export interface ThoughtEntry {
	type: 'THINKING' | 'TOOL';
	content: string;
}

File: webapp/backend/assistant/src/main/java/tom/assistant/controller/StreamingResponse.java
package tom.assistant.controller;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;

import tom.api.services.assistant.ChunkType;
import tom.api.services.assistant.LlmMetric;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record StreamingResponse(LlmStatus status, LlmMetric metric, List<String> sources, ChunkType type, String content) {

}
